libipt/src/pt_block_decoder.c

```diff
@@ -2515,7 +2521,14 @@ pt_blk_proceed_no_event_fill_cache(struct pt_block_decoder *decoder,
 	 * Cache updates are atomic so even if the two versions were not
 	 * identical, we wouldn't care because they are both correct.
 	 */
-	return pt_bcache_add(bcache, ioff, bce);
+	status = pt_bcache_add(bcache, ioff, bce);
+	if (status < 0)
+		return status;
+
+	if (valid_cache)
+		return pt_blk_proceed_no_event_cached(decoder, block,
+						bcache, msec);
```

Comment:
Please indent until the opening parenthesis using tabs (tab width 8) as far as possible, then switching to spaces.

Let's also add an empty line between the two returns.